jena-shex/src/main/java/org/apache/jena/shex/parser/ParserShExC.java

```diff
@@ -632,6 +641,48 @@ protected void numericFacetLength(String facetKind, int length, int line, int co
         addNodeConstraint(numLength);
     }
 
+    protected SemAct crackSemanticAction(String iriAndCode, int line, int column) {
+        // e.g. % <http://shex.io/extensions/Test/> { print(s) %}
+        // or   % ex:Test { print(s) %}
+        // or   %<http://shex.io/extensions/Test/>%
+
+        // Trim leading '%' and ws and trim trailing '%}'
+        String whitespaces = " \t\n\r\f";
+        int startOfIri = 1; // get past '%'
+        for (; whitespaces.indexOf(iriAndCode.charAt(startOfIri)) != -1; ++startOfIri)
```

Comment:
This all seems a bit brittle e.g. what if there's excess whitespace anywhere?

We have a commons-lang dependency already so `StringUtils.stripStart()` would be more robust

Reply:
I totally agree that it looks like 80's era K&R C, but I engineered this one pretty hard and I don't think that StringUtils will add anything but cycles. I [added tests](https://github.com/apache/jena/pull/1637/files#diff-f1b4d58d2a4e81046c490afc8b71496e8440f6aed8e15c3ded996bdb8de0c4c9) to verify that whatever the parser accepted, my C ptrs could handle.

Reply:
The code is for everyone to read ... including the future-eric who has rid himself of the 1980's.

Reply:
It adds clarity.